Basic style

mitogh opened the pull request.

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

mitogh committed:
Merge branch 'development'

orcunilbeyli committed:
Change <title> PHP Code

wp_title has some issues. Returns name of blog empty. Fixed this.

orcunilbeyli committed:
Initial editings for base stylement.

orcunilbeyli committed:
Did some changes on navigation

mitogh committed:
Revert "Component Style and Fixes"

This reverts commit 0f76a614459aae4d9284dfd5af2bc6ba49f82eaa.

mitogh committed:
Revert "Revert "Component Style and Fixes""

This reverts commit 95617d4973afe3d8cebeec5466014db00418fa45.

orcunilbeyli committed:
Menu Styles

Dropdown menu style with pure css/html.

orcunilbeyli committed:
Misc

orcunilbeyli committed:
Menu enhancements

Widen the menu wrapper to 100% and centerised the actual menu and some
color changes.

orcunilbeyli committed:
Component Style and Fixes

Comments section styled from scratch. Some fixes on header and footer.
Added new generated .scss files to app.scss. Overall responsive fixes in
responsive.scss. Added vatiables that used in fixes and new styles in
_variables.scss

orcunilbeyli committed:
Single page Nav Changes & Entry Info

Little changes on single page navigation and added margin to entry meta
data.

jrrrsmith commented on assets/sass/_app.scss:
```diff
@@ -40,9 +39,13 @@ Tags:
 3.0 Variables
 --------------------------------------------------------------*/
 @import "components/variables";
+@import "components/content";
```
@nucro the content, header, footer, and comments scss files should go in a folder named "modules" instead of "components".  And they should be listed in a new section 7.0.  please review this file for an example:
https://github.com/moxienyc/amtrak-journey/blob/develop/wp-content/themes/amtrak-journey/assets/sass/_app.scss

orcunilbeyli replied:
Yeah we talked about this with @mitogh . I'll fix it after design and other things are approved. @jrrrsmith

jrrrsmith commented on assets/sass/components/_content.scss:
```diff
@@ -0,0 +1,56 @@
+article {
```
@nucro please use the "@include fontSize" mixin for all font sizes.  please reference this file for an example:
https://github.com/moxienyc/amtrak-journey/blob/develop/wp-content/themes/amtrak-journey/assets/sass/modules/_blog-feed.scss#L43

jrrrsmith commented on assets/sass/components/_footer.scss:
```diff
@@ -0,0 +1,11 @@
+.site-footer {
+  font-size: 12px;
```
@nucro please use the fontSize mixin here as well

jrrrsmith commented on assets/sass/components/_header.scss:
```diff
@@ -0,0 +1,77 @@
+  & h1 {
+    padding  : 0;
+    margin   : 0;
+    font-size: 2rem;
```
@nucro again, please use the fontSize mixin

jrrrsmith commented on assets/sass/layouts/_responsive.scss:
```diff
@@ -0,0 +1,27 @@
+@media (max-width: 1130px) {
```
please create a variable for this media query in the _grid-settings.scss file here:
https://github.com/moxie-leean/Leean/blob/master/assets/sass/_grid-settings.scss

and use @include media($your-variable-name){
/\* code here */
}

here is an example: https://github.com/moxienyc/Printmornyc.com/blob/develop/wp-content/themes/Printmornyc/assets/sass/modules/_my-address.scss#L24

orcunilbeyli replied:
Oh okay. cool.

jrrrsmith commented on assets/sass/components/_header.scss:
```diff
@@ -0,0 +1,77 @@
+.site-header {
+  padding-top: 60px;
+}
+.site-title {
```
@nucro we should change the markup in the header so that we can follow the BEM guidelines more closely with our SCSS.  For example, the header has a class of "site-header", then there is a div in the header with the class "site-branding", a wrapper around the h1 with a class of "site-title", and then a div with a class named "site-description"  The "site-title" div around the h1 should be removed, and the format of the class names should be changed to this:

header class="header"
 div class="header__branding"
  h1 class="header__branding__title"
  div class="header__branding__description"

Then you can use BEM in your SCSS and the code would be like so:

.site-header{
  padding-top: 60px;
  &__branding{
  /\* styles here _/
    &__title{
      /_ h1 styles here _/
    }
    &__description{
      /_ .site-description styles here */
    }
  }
}

please see the header.php file here https://github.com/moxie-leean/Leean/blob/master/header.php

Let me know if you have any questions

cc/ @mitogh

jrrrsmith commented on assets/sass/components/comments.scss:
```diff
@@ -0,0 +1,27 @@
+.comments-area {
```
@nucro please edit the markup for the comments section also, add appropriate class names and then use them for the scss here following BEM and SMACCS.  You can reference the example that I left in the header.scss file.

jrrrsmith commented on assets/sass/layouts/_responsive.scss:
```diff
@@ -0,0 +1,27 @@
+  }
+
+  ul.menu {
+    li:hover ul {
```
@nucro please follow BEM here as well. You will need to edit the class names in the markup and then use those class names in the SCSS

mitogh replied:
Hey @nucro any feedback on this?

jrrrsmith commented on header.php:
```diff
@@ -14,7 +14,7 @@
+	<title><?php bloginfo('name'); ?> | <?php is_front_page() ? bloginfo('description') : wp_title(''); ?></title>
 	<link rel="profile" href="http://gmpg.org/xfn/11">
 	<link rel="pingback" href="<?php bloginfo( 'pingback_url' ); ?>">
 
```
@nucro please click on the "details" links for the Travis reports to see how you can fix the errors in the PHP files.  It's usually very simple changes, such as adding a space or removing an extra line.

you can see this at the bottom of the pull request here: https://github.com/moxie-leean/Leean/pull/50

<img width="880" alt="screen shot 2015-12-01 at 12 08 27 pm" src="https://cloud.githubusercontent.com/assets/7597286/11507934/5ea36b62-9824-11e5-9406-731786b8bbb5.png">

orcunilbeyli committed:
Moved Files from Comps to Modules

Moved comments, content, footer and header from Components folder to
Modules folder. Made fixes in _app according to changes.

mitogh commented:
@nucro some elements does not have any hover state, I think a change of the color using: 

- ```lighten``` or 
-  ``darken```

To update the color of the elements in order to give some visual feedback that we are doing over on the elements, please refer to this images. 

![screen shot 2015-12-01 at 12 03 46](https://cloud.githubusercontent.com/assets/3921289/11509459/cc74ccfe-9823-11e5-970c-f124316ed16f.png)
![screen shot 2015-12-01 at 12 03 48](https://cloud.githubusercontent.com/assets/3921289/11509460/cc788182-9823-11e5-8148-b7c524e49d42.png)

mitogh commented:
The line height inside of the captions should be smaller please test with this code: 

```html
<div id="attachment_906" style="width: 590px" class="wp-caption aligncenter"><img class="size-full wp-image-906  " title="Image Alignment 580x300" alt="Image Alignment 580x300" src="http://lean.dev/wp-content/uploads/2013/03/image-alignment-580x300.jpg" width="580" height="300"><p class="wp-caption-text">Look at 580×300 getting some <a title="Image Settings" href="http://en.support.wordpress.com/images/image-settings/">caption</a> love.</p></div>
```

On this element to be precise: 

```html
<p class="wp-caption-text">Look at 580×300 getting some <a title="Image Settings" href="http://en.support.wordpress.com/images/image-settings/">caption</a> love.</p>
```

Reference: 

![screen shot 2015-12-01 at 12 07 30](https://cloud.githubusercontent.com/assets/3921289/11509537/3fbc6bae-9824-11e5-8c36-7ffc0753af91.png)

mitogh commented:
Looks good so far @nucro great job, just take a look at the comments above provided by @jrrrsmith and I. 

Also I would like to know the code you used to recreate the menu, let me know if you have any other question or comments about the comments above. 

Thanks.

orcunilbeyli committed:
font-size to fontSize mixin

Changed font-size with fontSize mixin in content, footer and header scss
files.

orcunilbeyli committed:
Changed Responsive Trigger with Bourbon

Changed @media with @include media and added according settings to
_grid-settings.scss file.

orcunilbeyli committed:
Header Menu Fix

Somehow, .menu class should be tagged with ul not div. Anyways,
_header.scss changed according to that.

orcunilbeyli committed:
travis ci fix

Changed title tag with old one to fix travis ci issue.

orcunilbeyli committed:
BEM Sample HTML

Created sample html from production index from browser save. It’ll make
developer to see easier which classes and ids changed in one file. Only
for dev purposes.

orcunilbeyli committed:
Header tags to BEM

Changed only header element tags to BEM.

orcunilbeyli committed:
Main Area tags to BEM

Changed main and main>article area to BEM.

orcunilbeyli committed:
Footer tags to BEM

Footer elements and tegs changed to BEM.

mitogh commented:
@nucro do you have any feedback from the comments above?

https://github.com/moxie-leean/Leean/pull/50

Thanks.

nucro commented:
@mitogh Yeah I am still working on that. I was busy with other tasks that Wissam assigned to me yesterday. And started to switch BEM today. Created a sample html in latest commits. Jeremy made some feedbacks. I am going to fix them tomorrow.

mitogh commented:
No problem just to make sure we are on the same page, because I didn't see any previous comment from you on this PR. If you leave comments here it's better rather to have visibility rather than email because we can keep the historically track of each comment on in each line of code as well.

nucro commented:
Okay will be extra careful about this. Thanks.

orcunilbeyli committed:
Fixed BEM Issues

Fixed BEM problems according to Jeremy’s feedback.

nucro commented:
Did all feedbacks that Jeremy addressed. After you approve html sample about BEM changes, I'll implement this to whole project and jump to style issues that Cris mentioned. @jrrrsmith @mitogh

orcunilbeyli committed:
Classic tags to BEM migration

Changed whole project’s tags to BEM methodlogy.

mitogh commented:
Hey @nucro I'm going to open some issues, please make sure to address them one you have a change also let me know if yo have any question about my comments on each issue.

orcunilbeyli committed:
Fixing Latest Issues

Fixed:
 Removed transition on links.
 Page nav styling,
 Private password button styling,
 Hove state for buttons,
 Img captions lineheight removed.
 Hover states for links inside entries
 BEM migration fixes
 Other small bugfixes.

orcunilbeyli committed:
Page navigation fixes

- Flexbox causes mis-placing while there are no previous page.
Converted to float left/right.
- Removed bemsample.html

orcunilbeyli committed:
Primary Navigation Fixes

Fixed navigation issues caused by bem migration.

orcunilbeyli committed:
Bug Fixes

Addressed bugs have been fixed.
 -Pagination it's not aligned correctly
 -Content hover should have a transition
 -Button styles do not match

orcunilbeyli committed:
Fixed #59

Fixed button and link transitions to .4s.

orcunilbeyli committed:
Issue fix #59

*removed pushing effect.
*removed background color and changed color of next/prev buttons to
make them more link-ish.

mitogh commented on 404.php:
```diff
@@ -47,10 +47,10 @@
 					<?php
 					/* translators: %1$s: smiley */
 					$archive_content = '<p>' . sprintf( __( 'Try looking in the monthly archives. %1$s', 'lean' ), convert_smilies( ':)' ) ) . '</p>';
-					the_widget( 'WP_Widget_Archives', 'dropdown=1', "after_title=</h2>$archive_content" );
```
Hey @nucro don't change nothing of PHP, basic styles means only sass files, please keep things simple and focus only on 1 task at the time.

Thank you.

orcunilbeyli replied:
This one seems interesting. I didn't change this and almost didn't touch this file. Even this line is still there in my local environment. @mitogh

mitogh commented on assets/sass/_grid-settings.scss:
```diff
@@ -24,3 +24,6 @@ $large-screen: em(860);
 $medium-screen-up: new-breakpoint(min-width $medium-screen 4);
 $large-screen-up: new-breakpoint(min-width $large-screen 8);
+
+$mobile: 1140px;
```
@nucro mobile is 1140px width?

mitogh commented on assets/sass/_grid-settings.scss:
```diff
@@ -24,3 +24,6 @@ $large-screen: em(860);
 $large-screen-up: new-breakpoint(min-width $large-screen 8);
+
+$mobile: 1140px;
+$go-mobile: new-breakpoint(max-width $mobile $grid-columns);
```
@nucro  What this variable means? can you make your variables more meaningful.

Thank you

mitogh commented on assets/sass/base/_typography.scss:
```diff
@@ -48,7 +48,6 @@ p {
 }
 
 a {
-  @include transition(color 0.1s linear);
```
Hey @nucro can you put this transition back into the site.

Thanks.

mitogh commented on assets/sass/base/_wordpress.scss:
```diff
@@ -83,9 +83,8 @@ a {
-      line-height: 17;
-      margin: 0;
+    font-size: 12px;
+     margin: 0;
```
Can you make sure your lines are indented correctly. 

Thanks.

mitogh commented on assets/sass/components/_variables.scss:
```diff
@@ -1,6 +1,18 @@
 -------------------------------------------------------------- */
-$icon-color: #ff0000;
-$primary-color: #ff0000;
+$icon-color: #F0648B;
```
Can you update the variables name following this guidelines:  http://islovely.co/posts/how-to-name-color-variables-in-sass/

Thank you.

mitogh commented on search.php:
```diff
@@ -8,13 +8,13 @@
 
-		<header class="page-header">
-			<h1 class="page-title">
+		<header class="page__header">
```
Both elements does not have any relationship as they are named here can you make sure it's named properly.

mitogh commented on partials/content.php:
```diff
@@ -64,8 +64,11 @@
 		?>
 		</span>
 		<?php endif; ?>
+<!--
```
Can you remove this html comment ?

mitogh commented on partials/content.php:
```diff
@@ -10,19 +10,19 @@
 
 <?php tha_entry_before(); ?>
-<article id="post-<?php the_ID(); ?>" <?php post_class(); ?>
+<article class="entry" id="post-<?php the_ID(); ?>" <?php post_class(); ?>
```
can you remove this class and open an issue to add this since this will add 2 class elements to the div.

mitogh commented on partials/content.php:
```diff
@@ -35,7 +35,7 @@
 	</div>
 	<?php else : ?>
-	<div class="entry-content" itemprop="articleBody">
+	<div class="entry__content entry__a--transition" itemprop="articleBody">
```
Can you rename: `entry__a--transition` for something more meaningful thanks.

orcunilbeyli replied:
removing entry__a--transition. We already decided to go for transition for all. Makes sense? @mitogh

mitogh replied:
Ok makes more sense, so we are going to remove all `entry__a--transition.` right?

orcunilbeyli replied:
Yes.

mitogh commented on inc/class-theme-assets.php:
```diff
@@ -260,7 +260,7 @@ private function load_comments_assets() {
 
 			if ( $load_comments ) {
-				wp_enqueue_script( 'comment-reply' );
+				wp_enqueue_script( 'comment__reply' );
```
Same as before please don't update PHP code.

orcunilbeyli replied:
That was a mistaken change. Sorry reverting. @mitogh

mitogh replied:
OK please update all other areas where this is happening BEM applies only to HTML markup not PHP code.

mitogh commented on assets/sass/modules/_header.scss:
```diff
@@ -0,0 +1,74 @@
+  width: 100%;
+  background: #253645;
+}
+ul.menu {
```
Can you remove your styles for the menu?

Thanks.

orcunilbeyli replied:
What do you mean by this? Remove complete styling of main navigation? @mitogh

mitogh replied:
Yes please, current menu does not look ok and anyways in projects we are going to get ride of this menu.

orcunilbeyli replied:
This is how it looks now. It looks so ugly. Is it okay this way? Do you suggest any workaround for this? @mitogh 

http://snaps.getmoxied.net/1B1L1i3X1J42

mitogh replied:
We can work in a new feature in future just for the menu.

mitogh commented on assets/sass/modules/_header.scss:
```diff
@@ -0,0 +1,74 @@
+}
+.site-description {
+    @include fontSize(16px);
+    color:#c0c0c0;
```
Please for colours use variables

mitogh commented on assets/sass/modules/_footer.scss:
```diff
@@ -0,0 +1,14 @@
+  font-family: sans-serif;
+  padding: 25px 0px;
+  text-align: right;
+  & .footer__info a {
```
Can you indent this lines of code, thanks.

mitogh commented on assets/sass/modules/_content.scss:
```diff
@@ -0,0 +1,111 @@
+    font-family: sans-serif;
+    transition: 0.4s;
+    &:hover {
+      color:#fff;
```
Same use variables for colours thanks.

mitogh commented on assets/sass/modules/_comments.scss:
```diff
@@ -0,0 +1,23 @@
+    padding: 10px;
+    margin: 0;
+    outline: 0;
+    border: 2px solid white;
```
Try to follow 1 rule here you are using color names instead of using hex values, please use variables to references colours in the theme.

Thanks.

orcunilbeyli committed:
Pull request fixes

*added link transition back
*Indent fixes on wordpress.scss,content.scss
*color variable enchancements
*link/hover enchancements

orcunilbeyli committed:
Color Variable Name Enchancements

Added underscores to color names as prefix. Changed other variables
according to that.

orcunilbeyli committed:
PHP related BEM fixes

*reverted to old state the php codes that changed mistakenly.

orcunilbeyli committed:
Transition Removal

Removed experimental transition class.

orcunilbeyli committed:
Menu Style Removal

Removed dropdown menu style to naked.

orcunilbeyli committed:
Tag removal

Removed tags that has no impact on CSS side.

orcunilbeyli committed:
Variable Change to HEX

Some color variables are changed from colorname to hex.

mitogh committed:
Small fixes on Leean theme
